### PR TITLE
add `polars-distance`

### PR DESCRIPTION
**Pull request by moritzwilksch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/polars-distance/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set name = "polars-distance" %}
+{% set version = "0.4.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/polars_distance-{{ version }}.tar.gz
+  sha256: 9c0e185e91a10938a8a3b8fb894d6f6be0e0212b07db8539acfb15f7e79de93d
```
Comment (pavelzw) [inline recipes/polars-distance/meta.yaml:10]:
```suggestion
  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/polars_distance-{{ version }}.tar.gz
    sha256: 9c0e185e91a10938a8a3b8fb894d6f6be0e0212b07db8539acfb15f7e79de93d
  - url: https://raw.githubusercontent.com/ion-elgreco/polars-distance/main/LICENSE
    sha256: 017feb4ae87126410f1d9beb6af4c9df3c25a0051d87db66da6f26c509f5348d
```

File: recipes/polars-distance/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/ion-elgreco/polars-distance
+  summary: Polars plugin for pairwise distance functions
+  license: MIT
+  license_file: THIRDPARTY.yml
```
Comment (pavelzw) [inline recipes/polars-distance/meta.yaml:41]:
```suggestion
  license_file:
    - LICENSE
    - THIRDPARTY.yml
```

File: recipes/polars-distance/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - {{ compiler('rust') }}
+    - cargo-bundle-licenses
```
Comment (pavelzw) [inline recipes/polars-distance/meta.yaml:20]:
```suggestion
    - cargo-bundle-licenses
    - make
```

File: recipes/polars-distance/build.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+$PYTHON -m pip install . -vv
+cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
```
Comment (corneliusroemer) [inline recipes/polars-distance/build.sh:6]:
```suggestion
cargo-bundle-licenses --format yaml --output THIRDPARTY.yml

```

**Comment by moritzwilksch** [comment]
@conda-forge/help-python, ready for review!

**Review by xhochy — APPROVED** [review]

**Pull request merged**